### PR TITLE
1786 layout improvements

### LINKED ISSUE #1786: Add MinWidth and Wrap to Layout Attribute
Sometimes we want a bit extra control of how an item is showed.

- MinWidth: This is mostly for column based displays: It would be nice to specify an expected minimum width of the content, so that the column does not start out way too small.
- WordWrap: some controls are intended to use word wrapping and for some it is better to let the text overflow. We cannot figure this out based on the use, so we should let the developer decide.

### PR DESCRIPTION
**Pull request by rmadsen-ks**
Close #1786

### CODE REVIEW

**Review by alnlarsen — CHANGES_REQUESTED** [review]
See comments. Let's discuss!

File: Engine/UserInput.cs [hunk 1]
```diff
@@ -104,7 +107,11 @@ public class LayoutAttribute : Attribute
         /// <summary> Maximum row height for the input. </summary>
         public int MaxRowHeight { get; } = 1000;
 
-        /// <summary> </summary>
+        /// <summary> Minimum width in the width of an average character (approximately.).
+        /// This can be used for displays where an item is showed as a column where the width is a variable.</summary>
+        public int MinWidth { get; set; } = -1;
```
[the remark this answers is not in the document]
Comment (rmadsen-ks) [reply Engine/UserInput.cs]: I dont know. What is the default unit of MinWidth? We dont write the unit for height either.

File: Engine/UserInput.cs [hunk 2]
```diff
@@ -90,7 +90,10 @@ public enum LayoutMode
         /// <summary> The user input fills the whole row. </summary>
         FullRow = 2,
         /// <summary> The user input floats to the bottom.</summary>
-        FloatBottom = 4
+        FloatBottom = 4,
+        ///<summary> Whether word wrapping should be enabled. </summary>
+        WordWrapping = 8
```
[the remark this answers is not in the document]
Comment (rmadsen-ks) [reply Engine/UserInput.cs]: I dont think so. We should default to no wrapping.

**Review by alnlarsen — APPROVED** [review]
This looks great!

**Pull request merged**